Cocoaheads School_Lesson1.playground/Contents.swift

```diff
@@ -0,0 +1,93 @@
+var arrayNumbers = [Double]()
+
+func meanOfNumbers(_ arrayNumbers: [Double]) -> Double? {
+    if arrayNumbers.isEmpty {
+        return nil
+    }
+    var total: Double = 0
+    for number in arrayNumbers {
+        total += number
+    }
+    return total / Double(arrayNumbers.count)
+}
+
+if let mean = meanOfNumbers(arrayNumbers) {
+    print("The mean of numbers is: \(mean)")
+} else {
+    print("The array is empty")
+}
+
+func sortedArrayBubble(_ arrayNumbers: [Double]) -> [Double] {
+    guard arrayNumbers.count > 1 else {
+        return arrayNumbers
+    }
+    var outputArray = arrayNumbers
+    for index in 0..<arrayNumbers.count {
+        for seсIndex in 0..<((outputArray.count - 1)-index) {
+            if outputArray[seсIndex] > outputArray[seсIndex + 1] {
+                let temp = outputArray[seсIndex]
+                outputArray[seсIndex] = outputArray[seсIndex + 1]
+                outputArray[seсIndex + 1] = temp
+            }
+        }
+    }
+    return outputArray
+}
+
+func medianOfNumbers(_ arrayNumbers: [Double]) -> Double? {
+    if arrayNumbers.isEmpty {
+        return nil
+    }
+    let sortedArray = sortedArrayBubble(arrayNumbers)
+    let countOfArray = arrayNumbers.count
+    if countOfArray.isMultiple(of: 2) {
+        return Double(sortedArray[countOfArray/2] + sortedArray[countOfArray/2 - 1]) / 2.0
+    } else {
+        return Double(sortedArray[countOfArray/2])
+    }
+}
+
+if let median = medianOfNumbers(arrayNumbers) {
+    print("The median of numbers is: \(median)")
+} else {
+    print("The array is empty")
+}
+
+func modeOfNumbers(_ arrayNumbers: [Double]) -> [Double:Int]? {
+    guard arrayNumbers.count > 1 else {
+        return nil
+    }
+    let sortedArray = sortedArrayBubble(arrayNumbers)
+    var dicOfMode = [Double:Int]()
+    var count = 0
+    var modeCount = 0
+    var compareElement = sortedArray[0]
+    for index in 0..<sortedArray.count {
+        if sortedArray[index] == compareElement {
+           count += 1
+        } else {
+            if modeCount <= count && count>1 {
+                modeCount = count
+                dicOfMode.updateValue(modeCount, forKey: compareElement)
+            }
+            compareElement = sortedArray[index]
+            count = 1
+        }
+    }
+    if modeCount <= count && count>1 {
+        modeCount = count
+        dicOfMode.updateValue(modeCount, forKey: compareElement)
+    }
+    return dicOfMode
+}
+
+var mode = modeOfNumbers(arrayNumbers)
+if let dicOfMode = mode {
+   for modeNum in dicOfMode.keys {
```

Comment:
Try to avoid abbreviated words.

Make it clear and easy:

```suggestion
   for mode in dicOfMode.keys {
```

Reply:
done